File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int low = 0, high = height.size()-1;
        int maxA = 0;
        while(low < high){
            int area = (high-low) * min(height[low],height[high]);
            maxA = max(maxA,area);
            if(height[low] < height[high]){
                low++;
            }
            else high--;
        }
        return maxA;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: list[int]) -> int:
        l,r = 0,len(height)-1
        maxarea = 0
        while l<r:
            area = (r-l)*min(height[l],height[r])
            maxarea = max(maxarea,area)
            if height[l]>height[r]:
                r -= 1
            else:
                l += 1

        return maxarea